modules/planning/reference_line/reference_line.cc

```diff
@@ -521,8 +519,8 @@ bool ReferenceLine::IsOnLane(const SLBoundary& sl_boundary) const {
   double lane_left_width = 0.0;
   double lane_right_width = 0.0;
   map_path_.GetLaneWidth(middle_s, &lane_left_width, &lane_right_width);
-  return !(sl_boundary.start_l() > lane_left_width ||
-           sl_boundary.end_l() < -lane_right_width);
+  return !sl_boundary.start_l() > lane_left_width ||
+         !sl_boundary.end_l() < -lane_right_width;
```

Comment:
!(a || b) <=> !a && !b

Reply:
Also, ! has higher priority than > and <

Reply:
IMO, the best solution is "return sl_boundary.start_l() <= lane_left_width && sl_boundary.end_l() >= -lane_right_width".  In this way, we can both improve the code readability and get rid of the priority of operator issue. what's your opinion?

Reply:
I'm fine with the solution as long as the logic remains the same.

Reply:
@freeHackOfJeff Agree. Positive logic is more friendly. Could you refine it?

Reply:
Thanks both your feedback, @xiaoxq sure, will refine it ASAP. :)